src/modules/test/unit-tests/TestRecipeParserTests.cpp

```diff
@@ -25,4 +25,20 @@ namespace Tests
         EXPECT_STRCASEEQ("", testRecipes->at(1).m_payload.c_str());
         EXPECT_EQ(0, testRecipes->at(1).m_payloadSizeBytes);
     }
+
+    TEST(TestRecipeParserTests, RequiredFieldsMissing)
+    {
+        std::stringstream newBuffer;
+        std::streambuf *originalBuffer = std::cerr.rdbuf();
+        // Redirect cerr to our buffer
```

Comment:
>// Redirect cerr to our buffer [](http://example.com/codeflow?start=8&length=30)

Does this need to be reset once done or is self-reverting?

Reply:
Its not self-reverting which is why we save/restore the original stream at the end